dags/sentinel2_gap_report.py

```diff
@@ -0,0 +1,87 @@
+"""
+# Generate a gap report between sentinel-cogs and deafica-sentinel-2 buckets
+
+This DAG runs once a month and creates a gap report in the folowing location:
+s3://deafrica-sentinel-2/monthly-status-report
+"""
+import json
+import csv
+from pathlib import Path
+from datetime import datetime
+
+from airflow import configuration
+from airflow import DAG
+from airflow.operators.python_operator import PythonOperator
+from utils.inventory import s3
+
+default_args = {
+    'owner': 'Airflow',
+    "start_date": datetime(2020, 7, 24),
+    'email': ['toktam.ebadi@ga.gov.au'],
+    'email_on_failure': True,
+    'email_on_success': True,
+    'email_on_retry': False,
+    'retries': 0,
+    'manifest_suffix': "manifest.json",
+    'africa_tiles': "data/africa-mgrs-tiles.csv",
+    'africa_conn_id': "deafrica-prod-migration",
+    "us_conn_id": "deafrica-migration_us",
+    "dest_bucket_name": "s3://deafrica-sentinel-2-inventory",
+    "src_bucket_name": "s3://sentinel-cogs-inventory",
+    "reporting_bucket": "s3://deafrica-sentinel-2",
+    "reporting_prefix": "monthly-status-report/",
+    "schedule_interval": "@monthly"
+}
+
+def generate_bucket_diffs():
+    """
+    Compare Sentinel-2 buckets in US and Africa and detect differences
+    A report containing missing keys will be written to s3://deafrica-sentinel-2/monthly-status-report
+    """
+    url_source = default_args['src_bucket_name']
+    url_destination = default_args['dest_bucket_name']
+    suffix = default_args['manifest_suffix']
+    africa_tile_ids_path = tile_ids_filepath = Path(configuration.get('core', 'dags_folder')). \
```

Comment:
Do we need this `tile_ids_filepath` variable?

Reply:
It was redundant, deleted.